utils.py

```diff
@@ -1,11 +1,11 @@
 
-def EQUAL(a, b):
+def sql_equal(a, b):
     return a == b or ((a != a) and (b != b))
 
 
-def ARRAYEQUAL(x, y):
+def array_equal(x, y):
```

Comment:
Unless there's a good reason I'm missing, function names should not be capitalised, and they should be snake_cased.